osu.Game.Rulesets.Osu/Objects/Drawables/DrawableOsuHitObject.cs

```diff
@@ -45,7 +45,7 @@ protected override void SkinChanged(ISkinSource skin, bool allowFallback)
         {
             base.SkinChanged(skin, allowFallback);
 
-            if (HitObject is IHasComboInformation combo)
+            if (HitObject is IHasComboInformation combo && this.HitObject.HitActions.Length != 1)
```

Comment:
Yeah, I wasn't a big fan of messing with the code of unrelated classes, and tried to keep it to a minimum, but just didn't see how else it could be done because the hitobjects don't natively support any distinction between the actions that triggered them. Maybe I can make an extended class with some overrides for the mod and repopulate the beatmap with them. I'll look into that sometime.

Reply:
@peppy please given them a direction on how to tackle this otherwise. @zachllogan you should instead expose some sort of action for `IApplicableToDrawableHitObject` to bind to and apply the colour.

Reply:
@smoogipoo I'll look into this when I have time. Also, sorry for sloppy code and taking up your time. I wrote this maybe 9 moths ago when I first touched the code, and there is no documentation, so exploring the code and modifying it is really the only way to get familiar (it's really well made & organized though, btw, which makes it way easier).

Reply:
In thinking this over a bit more we already expose the `ApplyCustomUpdateState` event which you should be able to use.